packages/infra/config/ihe-gateway-config.ts

```diff
@@ -0,0 +1,64 @@
+import { Duration } from "aws-cdk-lib";
+import { ECS_CPU, ECS_MEMORY } from "./aws/ecs";
+import { RDSAlarmThresholds } from "./aws/rds";
+
+export type IHEGatewayProps = {
+  vpcId: string;
+  certArn: string;
+  subdomain: string; // Subdomain for IHE integrations
+  /**
+   * ID of the existing private hosted zone where the IHE Gateway will be deployed.
+   */
+  privateZoneId: string;
+  ecs: {
+    // Watch out for the combination of vCPUs and memory, more vCPU requires more memory
+    // https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#task_size
+    /**
+     * Determines the valid CPU values.
+     */
+    memory: ECS_MEMORY;
+    /**
+     * Determined by the memory value.
+     */
+    cpu: ECS_CPU;
+    /**
+     * Minimum amount of tasks to run at all times.
+     */
+    minCapacity: number;
+    /**
+     * Maximum amount of tasks to run at all times.
+     */
+    maxCapacity: number;
+    /**
+     * How long a request can take before it is closed by the Load Balancer.
+     */
+    maxRequestTimeout: Duration;
+  };
+  java: {
+    /**
+     * Java's `-Xmx` value. https://eclipse.dev/openj9/docs/xms/
+     */
+    xmx: string;
+    /**
+     * Java's `-Xms` value. https://eclipse.dev/openj9/docs/xms/
+     */
+    xms: string;
```

Comment:
nit: I know it's what the config is for, but would it be useful to be explicit in the config?

ie changing to `initialHeapSize` and `maxHeapSize`, respectively, would avoid someone looking at the config wondering wtf those settings mean

Reply:
updated